packages/nimble-components/src/table/testing/table.pageobject.ts

```diff
@@ -447,6 +448,58 @@ export class TablePageObject<T extends TableRecord> {
         }
     }
 
+    /**
+     * @param columnIndex The index of the column to the left of a divider being dragged. Thus, this
+     * can not be given a value representing the last column index.
+     * @param deltas The series of mouse movements in the x-direction while sizing a column.
+     */
+    public dragSizeColumn(columnIndex: number, deltas: number[]): void {
+        const divider = this.getColumnDivider(columnIndex);
+        divider.setAttribute('active', 'true');
+        const dividerRect = divider.getBoundingClientRect();
+        const mouseDownEvent = new MouseEvent('mousedown', {
+            clientX: (dividerRect.x + dividerRect.width) / 2,
+            clientY: (dividerRect.y + dividerRect.height) / 2
+        });
+        divider.dispatchEvent(mouseDownEvent);
+
+        for (const delta of deltas) {
+            const mouseMoveEvent = new MouseEvent('mousemove', {
+                movementX: delta
+            });
+            document.dispatchEvent(mouseMoveEvent);
+        }
+
+        const mouseUpEvent = new MouseEvent('mouseup');
+        document.dispatchEvent(mouseUpEvent);
+    }
+
+    public getColumnDivider(index: number): HTMLElement {
+        const headerContainers = this.tableElement.shadowRoot!.querySelectorAll('.header-container');
+
+        if (index >= headerContainers.length - 1) {
+            throw new Error(
+                'Attempting to index past the number of available dividers. Max index is two less than the number of columns.'
```

Comment:
Would it be better to write this to return `HTMLElement | null` so that you only throw if you are trying to index past the number of columns, not the number of dividers. Then you can write tests that the right-most column doesn't have a divider on its right -- which currently is a bug in the code based on where the hidden columns fall within the DOM.